### PR TITLE
feat: add json output to `boostd pieces` command

### PR DESCRIPTION
**Pull request by LexLuthr**

### CODE REVIEW

File: cmd/boostd/pieces.go [hunk 1]
```diff
@@ -109,8 +136,34 @@ var piecesListCidInfosCmd = &cli.Command{
 						"DealOffset":  deal.Offset,
 						"DealLen":     deal.Length,
 					})
+
+					deal := map[string]interface{}{
+						"Deal":       deal.DealID,
+						"Sector":     deal.SectorID,
+						"DealOffset": deal.Offset,
+						"DealLen":    deal.Length,
```
Comment (dirkmc) [inline cmd/boostd/pieces.go:144]:
```suggestion
						"ID":       deal.DealID,
						"Sector":     deal.SectorID,
						"Offset": deal.Offset,
						"Length":    deal.Length,
```

File: cmd/boostd/pieces.go [hunk 2]
```diff
@@ -109,8 +136,34 @@ var piecesListCidInfosCmd = &cli.Command{
...
+					}
+					tpbl := pbl{
+						"PieceCid":    pi.PieceCID,
+						"BlockOffset": location.RelOffset,
+						"BlockLen":    location.BlockSize,
```
Comment (dirkmc) [inline cmd/boostd/pieces.go:149]:
```suggestion
						"BlockLength":    location.BlockSize,
```

File: cmd/boostd/pieces.go [hunk 3]
```diff
@@ -146,6 +199,28 @@ var piecesInfoCmd = &cli.Command{
...
+					"DealID":   d.DealID,
+					"SectorID": d.SectorID,
+					"Offset":   d.Offset,
+					"Length":   d.Length,
+				}
+				deals = append(deals, dl)
+			}
+
+			pieceinfo := map[string]interface{}{
+				"pieceCid": pi.PieceCID,
```
Comment (dirkmc) [inline cmd/boostd/pieces.go:217]:
```suggestion
				"PieceCid": pi.PieceCID,
```

File: cmd/boostd/pieces.go [hunk 4]
```diff
@@ -146,6 +199,28 @@ var piecesInfoCmd = &cli.Command{
 			return err
 		}
 
+		if cctx.Bool("json") {
+
+			type deal map[string]interface{}
+			var deals []deal
+			for _, d := range pi.Deals {
+				dl := deal{
+					"DealID":   d.DealID,
```
Comment (dirkmc) [inline cmd/boostd/pieces.go:208]:
```suggestion
					"ID":   d.DealID,
```

**Review by dirkmc — APPROVED** [review]
LGTM 👍

**Comment by dirkmc** [comment]
@LexLuthr could you please paste sample output for each command.
(feel free to do this tomorrow)

**Comment by LexLuthr** [comment]
Sample Output:


```
miner@worker-gpu-2:~$ boostd --json pieces list-cids
{
  "dataCids": [
    {
      "/": "bafykbzacea4f4ewfkomqb2wybbcqjo4pw6nu6rqkaaxvy6x7ylx27b6o3l4xa"
    },
    {
      "/": "bafykbzaceaw46ydxxo7otaqyhvkntcikunbkq4gt2uyafg4376a6bfbk5ycjs"
    },
    {
      "/": "bafykbzaceb33w74tnqlhnoshlgvti44wkpgumb4i44lqqk44bhb2i72hxhshu"
    },
    {
      "/": "bafykbzacebrxvigmhzxnuo7gywzz2ugxmq3nxywdg577ybxh2wd5lud5e7ebi"
    },
    {
      "/": "bafykbzaced6z4qy3p7zuyrzivef4ly5x5mfhd67daevlwzlsrxqvwqnet7ike"
    }
  ]
}
miner@worker-gpu-2:~$ boostd --json pieces list-cids -v
[
  {
    "DataCid": {
      "/": "bafykbzacea4f4ewfkomqb2wybbcqjo4pw6nu6rqkaaxvy6x7ylx27b6o3l4xa"
    },
    "PieceBlockLocation": [
      {
        "BlockLength": 0,
        "BlockOffset": 0,
        "Deal": {
          "ID": 102670,
          "Length": 268435456,
          "Offset": 0,
          "Sector": 27
        },
        "PieceCid": {
          "/": "baga6ea4seaqa7qvwz5sdzcsfzytiix6btpmdj2fq7oo2pmqe3d7x3mux4oaqeji"
        }
      }
    ]
  },
  {
    "DataCid": {
      "/": "bafykbzaceaw46ydxxo7otaqyhvkntcikunbkq4gt2uyafg4376a6bfbk5ycjs"
    },
    "PieceBlockLocation": [
      {
        "BlockLength": 0,
        "BlockOffset": 0,
        "Deal": {
          "ID": 167812,
          "Length": 2097152,
          "Offset": 0,
          "Sector": 35
        },
        "PieceCid": {
          "/": "baga6ea4seaqg3f6kdxkfg2uehyjc7ai2n3jaekidklr6zl5iovhigplah3yesaa"
        }
      },
      {
        "BlockLength": 0,
        "BlockOffset": 0,
        "Deal": {
          "ID": 175256,
          "Length": 2097152,
          "Offset": 536870912,
          "Sector": 40
        },
        "PieceCid": {
          "/": "baga6ea4seaqg3f6kdxkfg2uehyjc7ai2n3jaekidklr6zl5iovhigplah3yesaa"
        }
      }
    ]
  },
  {
    "DataCid": {
      "/": "bafykbzaceb33w74tnqlhnoshlgvti44wkpgumb4i44lqqk44bhb2i72hxhshu"
    },
    "PieceBlockLocation": [
      {
        "BlockLength": 0,
        "BlockOffset": 0,
        "Deal": {
          "ID": 103505,
          "Length": 8589934592,
          "Offset": 0,
          "Sector": 30
        },
        "PieceCid": {
          "/": "baga6ea4seaqptl3dse2ologujiwdybfl4x3kfe3ifwm45xriuiv2iivv4jkbeai"
        }
      }
    ]
  },
  {
    "DataCid": {
      "/": "bafykbzacebrxvigmhzxnuo7gywzz2ugxmq3nxywdg577ybxh2wd5lud5e7ebi"
    },
    "PieceBlockLocation": [
      {
        "BlockLength": 0,
        "BlockOffset": 0,
        "Deal": {
          "ID": 103509,
          "Length": 8589934592,
          "Offset": 0,
          "Sector": 0
        },
        "PieceCid": {
          "/": "baga6ea4seaqjlkzav3tvsowinpohcnhnbukzh7ernl4p3735fophgeszossy6kq"
        }
      }
    ]
  },
  {
    "DataCid": {
      "/": "bafykbzaced6z4qy3p7zuyrzivef4ly5x5mfhd67daevlwzlsrxqvwqnet7ike"
    },
    "PieceBlockLocation": [
      {
        "BlockLength": 0,
        "BlockOffset": 0,
        "Deal": {
          "ID": 167811,
          "Length": 536870912,
          "Offset": 536870912,
          "Sector": 35
        },
        "PieceCid": {
          "/": "baga6ea4seaqbljxgkl6ojd35jyqsrac7s46zv22wa6pvwvxap675dke6vcmnyda"
        }
      },
      {
        "BlockLength": 0,
        "BlockOffset": 0,
        "Deal": {
          "ID": 175255,
          "Length": 536870912,
          "Offset": 0,
          "Sector": 40
        },
        "PieceCid": {
          "/": "baga6ea4seaqbljxgkl6ojd35jyqsrac7s46zv22wa6pvwvxap675dke6vcmnyda"
        }
      },
      {
        "BlockLength": 0,
        "BlockOffset": 0,
        "Deal": {
          "ID": 179795,
          "Length": 536870912,
          "Offset": 0,
          "Sector": 41
        },
        "PieceCid": {
          "/": "baga6ea4seaqbljxgkl6ojd35jyqsrac7s46zv22wa6pvwvxap675dke6vcmnyda"
        }
      }
    ]
  }
]
miner@worker-gpu-2:~$ boostd --json pieces list-pieces
{
  "pieceCids": [
    {
      "/": "baga6ea4seaqa7qvwz5sdzcsfzytiix6btpmdj2fq7oo2pmqe3d7x3mux4oaqeji"
    },
    {
      "/": "baga6ea4seaqapj7crd2xf6sqg36dn5ttx4s3ixn3igilhwy4wwd2nbceorfdsni"
    },
    {
      "/": "baga6ea4seaqaprayexhzm7zucdwiphycnt3gqbzxhb36afh54othjdrl7x2iioa"
    },
    {
      "/": "baga6ea4seaqbljxgkl6ojd35jyqsrac7s46zv22wa6pvwvxap675dke6vcmnyda"
    },
    {
      "/": "baga6ea4seaqctiuwpnrmhx5lg6k4n253dqpkcscanhn6aegjiaeffjbarg5jsfy"
    },
    {
      "/": "baga6ea4seaqdo6tycy4ejqhwow7kcmlpnf6ufzmaav2lhucccfjd7l7554m6oki"
    },
    {
      "/": "baga6ea4seaqg3f6kdxkfg2uehyjc7ai2n3jaekidklr6zl5iovhigplah3yesaa"
    },
    {
      "/": "baga6ea4seaqgdb3nt7eqzzclujypbay5ymqyqhrltv73r2euahdjdgczczv4una"
    },
    {
      "/": "baga6ea4seaqh5prrl6ykov4t64k6m5giijsc44dcxtdnzsp4izjakqhs7twauiq"
    },
    {
      "/": "baga6ea4seaqjlkzav3tvsowinpohcnhnbukzh7ernl4p3735fophgeszossy6kq"
    },
    {
      "/": "baga6ea4seaqomjbaafkaqm4uxk2qqhju75nl7sxbvzcmkyhyls6pxkuk6podsdi"
    },
    {
      "/": "baga6ea4seaqptl3dse2ologujiwdybfl4x3kfe3ifwm45xriuiv2iivv4jkbeai"
    }
  ]
}

miner@worker-gpu-2:~$ boostd --json pieces piece-info baga6ea4seaqptl3dse2ologujiwdybfl4x3kfe3ifwm45xriuiv2iivv4jkbeai
{
  "Deals": [
    {
      "ID": 103505,
      "Length": 8589934592,
      "Offset": 0,
      "SectorID": 30
    }
  ],
  "PieceCid": {
    "/": "baga6ea4seaqptl3dse2ologujiwdybfl4x3kfe3ifwm45xriuiv2iivv4jkbeai"
  }
}
miner@worker-gpu-2:~$ boostd --json pieces cid-info bafykbzaced6z4qy3p7zuyrzivef4ly5x5mfhd67daevlwzlsrxqvwqnet7ike
{
  "DataCid": {
    "/": "bafykbzaced6z4qy3p7zuyrzivef4ly5x5mfhd67daevlwzlsrxqvwqnet7ike"
  },
  "PieceBlockLocation": [
    {
      "BlockLocation": {
        "BlockSize": 0,
        "RelOffset": 0
      },
      "PieceCid": {
        "/": "baga6ea4seaqbljxgkl6ojd35jyqsrac7s46zv22wa6pvwvxap675dke6vcmnyda"
      }
    }
  ]
}

```

**Pull request merged**